test/unittests/evm_eip4844_blobhash_test.cpp

```diff
@@ -0,0 +1,92 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2023 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+/// This file contains EVM unit tests for the BLOBHASH instruction from EIP-4844
+/// https://eips.ethereum.org/EIPS/eip-4844
+
+#include "evm_fixture.hpp"
+
+using namespace evmc::literals;
+using evmone::test::evm;
+
+TEST_P(evm, blobhash_undefined)
+{
+    rev = EVMC_SHANGHAI;
+    execute(blobhash(0));
+    EXPECT_STATUS(EVMC_UNDEFINED_INSTRUCTION);
+}
+
+TEST_P(evm, blobhash_empty)
+{
+    rev = EVMC_CANCUN;
+    execute(blobhash(0) + ret_top());
+    EXPECT_OUTPUT_INT(0);
+
+    execute(blobhash(1) + ret_top());
+    EXPECT_OUTPUT_INT(0);
+}
```

Comment:
You could also test it with `0xff...ff`, here and in other tests

Reply:
Added.

Reply:
What about other tests?

Reply:
Added to all.